dlc/src/lib.rs

```diff
@@ -0,0 +1,967 @@
+};
+use secp256k1::ecdsa_adaptor::{AdaptorProof, AdaptorSignature};
+use secp256k1::schnorrsig::{PublicKey as SchnorrPublicKey, Signature as SchnorrSignature};
+use secp256k1::{Message, PublicKey, Secp256k1, SecretKey, Signature, Verification};
+
+pub mod util;
+
+/// Minimum value that can be included in a transaction output. Under this value,
+/// outputs are discarded
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#change-outputs
+const DUST_LIMIT: u64 = 1000;
+
+/// The transaction version
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#funding-transaction
+const TX_VERSION: i32 = 2;
+
+/// The base weight of a fund transaction
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#fees
+const FUND_TX_BASE_WEIGHT: usize = 214;
+
+/// The weight of a CET excluding payout outputs
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#fees
+const CET_BASE_WEIGHT: usize = 500;
+
+/// The base weight of a transaction input computed as: (outpoint(36) + sequence(4) + scriptPubKeySize(1)) * 4
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#fees
+const TX_INPUT_BASE_WEIGHT: usize = 164;
+
+/// The witness size of a P2WPKH input
+/// See: https://github.com/discreetlogcontracts/dlcspecs/blob/master/Transactions.md#fees
+pub const P2WPKH_WITNESS_SIZE: usize = 108;
+
+// Setting the nSequence for every input of a transaction to this value disables
+// both RBF and nLockTime usage.
+const DISABLE_LOCKTIME: u32 = 0xffffffff;
+// Setting the nSequence for every input of a transaction to this value disables
+// RBF but enables nLockTime usage.
+const ENABLE_LOCKTIME: u32 = 0xfffffffe;
+
+// TODO(tibo): rename local and remote to whatever is agreed upon in the specs.
+/// Represents the payouts for a unique contract outcome. Local party represents
+/// the initiator of the contract while remote party represents the party
+/// accepting the contract.
+pub struct Payout {
+    /// Payout for the local party
+    pub local: u64,
+    /// Payout for the remote party
+    pub remote: u64,
+}
+
+/// Contains the necessary transactions for establishing a DLC
+pub struct DlcTransactions {
+    /// The fund transaction locking both parties collaterals
+    pub fund: Transaction,
+    /// The contract execution transactions for closing the contract on a
+    /// certain outcome
+    pub cets: Vec<Transaction>,
+    /// The refund transaction for returning the collateral for each party in
+    /// case of an oracle misbehavior
+    pub refund: Transaction,
+}
+
+/// Contains info about a utxo used for funding a DLC contract
+pub struct TxInputInfo {
+    /// The outpoint for the utxo
+    pub outpoint: OutPoint,
+    /// The maximum witness length
+    pub max_witness_len: usize,
+    /// The redeem script
+    pub redeem_script: Script,
+}
```

Comment:
I'm not sure what I would do with the extra parameters? To me the prevTx validation is something that should happen at a higher layer.

Reply:
I see, well in that case (which I'm fine with) I still think the redeem script is meant to be an option?

Reply:
Mmmm at the moment I'm using an empty script instead of an option. IIRC the reason is that in the spec in the `FundingInput` type there is redeem script which can be an empty script IIUC. I could of course change it to an option but I'm not sure I see a big advantage?